packages/x-charts/src/internals/components/ChartsAxesGradients/ChartsAxesGradients.tsx

```diff
@@ -4,11 +4,23 @@ import { useChartId, useDrawingArea } from '../../../hooks';
 import ChartsPiecewiseGradient from './ChartsPiecewiseGradient';
 import ChartsContinuousGradient from './ChartsContinuousGradient';
 import { AxisId } from '../../../models/axis';
+import ChartsContinuousGradientObjectBound from './ChartsContinuousGradientObjectBound';
+import { useZAxis } from '../../../hooks/useZAxis';
 
 export function useChartGradient() {
   const chartId = useChartId();
   return React.useCallback(
-    (axisId: AxisId, direction: 'x' | 'y') => `${chartId}-gradient-${direction}-${axisId}`,
+    (axisId: AxisId, direction: 'x' | 'y' | 'z') => `${chartId}-gradient-${direction}-${axisId}`,
+    [chartId],
+  );
+}
+
+// TODO: make public?
+export function useChartGradientObjectBound() {
```

Comment:
Should both this and `useChartGradient` be made public? 

The gradients are not generated inside the `ContinuousColorLegend` anymore, instead, any necessary gradient is generated here, and the other parts of the charts will simply use them. Hence we need to generate both the regular gradient as well as an `objectBound` version of it, so it displays correctly on the legend, but also any mark or label we want to put it in.